gdp/column/gdp_column.py

```diff
@@ -47,178 +53,145 @@ def build_column(min_trays, max_trays, xD, xB):
-            m.vap[m.condens_tray - 1] - m.vap[m.condens_tray])
+        return m.dis == (1 - m.reflux_frac) * (m.vap[m.condens_tray - 1] - m.vap[m.condens_tray])
 
+    # Constraint to ensure the sum of component liquid flows from each tray equals the total liquid flow from that tray
     @m.Constraint(m.trays)
     def liquid_sum(m, t):
         return sum(m.L[c, t] for c in m.comps) == m.liq[t]
 
+    # Constraint to ensure the sum of component vapor flows from each tray equals the total vapor flow from that tray
     @m.Constraint(m.trays)
     def vapor_sum(m, t):
         return sum(m.V[c, t] for c in m.comps) == m.vap[t]
 
-    m.bottoms_sum = Constraint(
-        expr=sum(m.B[c] for c in m.comps) == m.bot)
-    m.distil_sum = Constraint(
-        expr=sum(m.D[c] for c in m.comps) == m.dis)
+    # Constraint to ensure the sum of component bottoms flows equals the total bottoms flow
+    m.bottoms_sum = Constraint(expr=sum(m.B[c] for c in m.comps) == m.bot)
+
+    # Constraint to ensure the sum of component distillate flows equals the total distillate flow
+    m.distil_sum = Constraint(expr=sum(m.D[c] for c in m.comps) == m.dis)
 
+    # Constraint to ensure that the temperature on each tray is greater or equal to the temperature on the tray below
     @m.Constraint(m.trays)
     def monotonoic_temperature(_, t):
         return m.T[t] >= m.T[t + 1] if t < max_trays else Constraint.Skip
 
+
+    # Building phase equilibrium for each conditional tray
     for t in m.conditional_trays:
         _build_tray_phase_equilibrium(m, t, m.tray[t])
+
+    # Defining a block for feed tray, reboiler and condenser phase equilibrium
     m.feed_tray_phase_eq = Block()
     m.reboiler_phase_eq = Block()
     m.condenser_phase_eq = Block()
+
+    # Building phase equilibrium for feed tray, reboiler and condenser
     _build_tray_phase_equilibrium(m, m.feed_tray, m.feed_tray_phase_eq)
     _build_tray_phase_equilibrium(m, m.reboil_tray, m.reboiler_phase_eq)
     _build_tray_phase_equilibrium(m, m.condens_tray, m.condenser_phase_eq)
+
+    # Building heat relations for the column
     _build_column_heat_relations(m)
 
+    # Constraint to ensure the distillate contains at least a certain purity of benzene
     @m.Constraint()
     def distillate_req(m):
         return m.D['benzene'] >= m.distillate_purity * m.dis
 
+    # Constraint to ensure the bottoms contains at least a certain purity of toluene
     @m.Constraint()
     def bottoms_req(m):
         return m.B['toluene'] >= m.bottoms_purity * m.bot
 
-    m.obj = Objective(
-        expr=(m.Qc + m.Qb) * 1E3 + 1E3 * (
-            sum(m.tray[t].indicator_var for t in m.conditional_trays) + 1),
-        sense=minimize)
+    # Define the objective function as the sum of reboiler and condenser duty plus an indicator for tray activation 
+    m.obj = Objective(expr=(m.Qc + m.Qb) * 1E3 + 1E3 * (sum(m.tray[t].indicator_var for t in m.conditional_trays) + 1), sense=minimize)
```

Comment:
Reading this line, don't you wonder where these factors are coming? What are the units of these constraints? Add them to the comments, please.

Reply:
Add the description of the phase equilibrium for each specific trays.

Reply:
So please add the units! In fact, this line defines an objective function as the sum of the reboiler and condenser heat times some constant (1e3) plus the soum of all the active trays, almost times a function 1e3. What units do the 1e3 have?